yt/visualization/volume_rendering/lens.py

```diff
@@ -819,9 +821,67 @@ def set_viewpoint(self, camera):
         """
         self.viewpoint = camera.position
 
+
+class HEALPixLens(Lens):
+    def __init__(self):
+        super(HEALPixLens, self).__init__()
+        self.nside = 1
+        self.radius = 1.0
+    
+    @property
+    def hp(self):
+        return HEALPix(self.nside)
+
+    def setup_box_properties(self, camera):
+        """Set up the view and stage based on the properties of the camera."""
+        self.radius = camera.width.max()
+        super(HEALPixLens, self).setup_box_properties(camera)
+        self.set_viewpoint(camera)
+    
+    def new_image(self, camera):
+        """Initialize a new ImageArray to be used with this lens."""
+        self.current_image = ImageArray(
+            np.zeros((self.hp.npix, 1, 4), dtype='float64'),
+            info={'imtype': 'rendering'})
+        return self.current_image
+
+    def _get_sampler_params(self, camera, render_source):
+        hp = self.hp
+
+        positions = np.ones((hp.npix, 1, 3), dtype='float64') * camera.position
```

Comment:
In old_camera/HEALpixCamera, there's an additional step:

```python
dx = min(g.dds.min() for g in self.ds.index.find_point(self.center)[0])
positions += self.inner_radius * dx * vs
```
where `vs` is the `vectors` below.

I dropped this part here because I'm confused about `dx`. What's its use? If my understanding is correct, `inner_radius` is defined in the unit of `dx`. But what's the reason behind it?

Reply:
I don't really have a lot of context here, but it looks like that snippet originated in this commit:

https://github.com/yt-project/yt/commit/252c8c4d0d48afd242fae7a77dc567d05120b536

Reply:
Thanks! This clears things up. My git ability needs much improvement. I've been struggling to find this information.

Reply:
Yeah, it's a little tough because of the renaming from `camera.py` to `old_camera.py`.